pyomo/core/tests/unit/test_con.py

```diff
@@ -134,9 +168,11 @@ def rule(model):
         model.c = Constraint(rule=rule)
 
         self.assertEqual(model.c.equality, False)
-        self.assertEqual(model.c.lower, None)
+        self.assertEqual(model.c.lower, float('-inf'))
```

Comment:
Is this change backwards compatible? Or were we testing for the existence of a bug?

Reply:
We were testing for the existence of a bug:  By (current) convention, `lower` returns the *expression* for the lower bound and `lb` returns the *value* (or `None` if not specified or `-inf`).  The problem was that in the old world order, some bounds processing was done "early" in `set_value` -- including some of the conversion from +/-inf to `None`.  We now defer that processing until the user asks for the value.  So, in this test, since the user provided the "expression" `float('-inf')` for the lower bound, the "correct" thing to do is return that expression from `lower`.